src/elements/EditableTable.tsx

```diff
@@ -0,0 +1,262 @@
+    return sum / numericVals.length || 0;
+};
+/**
+ * A table form element.
+ * @displayName Editable Table
+ * @description A table form element that allows editing.
+ * @param props The props that will be provided by the Workflow runtime.
+ */
+function EditableTableElement(props: EditableTableElementProps): React.ReactElement | null {
+    const {
+        rows,
+        cols,
+        maxHeight,
+        maxWidth,
+        onMouseEnter,
+        onMouseLeave,
+        rowCalculation,
+        raiseEvent,
+        setValue,
+        size,
+        stickyHeader,
+        footerLabel,
+    } = props;
+    const [data, setData] = useState(rows);
+    setValue(data);
+    useEffect(() => {
+        setData(rows);
+    }, [rows]);
+
+    const illegalCol = cols.some((x) => ["text", "number", "date", "time"].indexOf(x.type) == -1);
+    if (illegalCol) {
+        throw new Error("Unsupported column.  Column must be of type: text, number, date or time.");
+    }
+    const includeFooter = cols.some((x) => x.columnCalculation);
+    const footerRow: FooterRow = calculateFooterRow(cols, rows);
+
+    const handleMouseEnter = (event: React.MouseEvent, row: Row, column: string) => {
+        raiseEvent("custom", {
+            eventType: "mouseEnter",
+            rowData: { rows: data, row: row, column: column },
+        });
+        onMouseEnter?.(row);
+    };
+
+    const handleMouseLeave = (event: React.MouseEvent, row: Row, column: string) => {
+        raiseEvent("custom", {
+            eventType: "mouseLeave",
+            rowData: { rows: data, row: row, column: column },
+        });
+        onMouseLeave?.(row);
+    };
+
+    const handleCellChange = (
+        event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+        row: Row,
+        column: Column
+    ) => {
+        if (column?.parse) {
+            row[column.name] = column.parse(event, row, column);
+        } else if ((event.target as any).type === "number") {
+            row[column.name] = Number(event.target.value);
+        } else {
+            row[column.name] = event.target.value;
+        }
+        if (rowCalculation) {
+            rowCalculation(data, row);
+        }
+        raiseEvent("custom", {
+            eventType: "cellChange",
+            rowData: { rows: data, row: row, column: column },
+        });
+        setData([...data]);
+        setValue(data);
+    };
+
+    return (
+        <Box maxHeight={maxHeight} maxWidth={maxWidth} sx={{ overflow: "auto" }}>
+            <Table
+                size={size}
+                stickyHeader={stickyHeader}
+                sx={{
+                    // Hack required to override .gcx-forms.defaults
+                    ["& span"]: {
+                        display: "inline-flex !important",
+                    },
+                }}
+            >
+                <TableHead>
+                    <TableRow>
+                        {includeFooter && <TableCell />}
```

Comment:
Unless there is a `footerLabel` this extra column looks weird.

Reply:
Added a test for footerLabel